src/treetensornetworks/projttns/projttn.jl

```diff
@@ -3,15 +3,16 @@ ProjTTN
 """
 struct ProjTTN{V} <: AbstractProjTTN{V}
   pos::Union{Vector{<:V},NamedEdge{V}} # TODO: cleanest way to specify effective Hamiltonian position?
-  H::TTN{V}
+  operator::AbstractTTN{V}
```

Comment:
In general it isn't a good idea for performance to put an abstract type as a field. Is `AbstractTTN` needed here? If not I would just put it back to `TTN` and we can generalize it if needed (by making `ProjTTN` parameterized by the type of the `operator` field).